libs/accounts/email-sender/src/email-sender.ts

```diff
@@ -0,0 +1,276 @@
+      secure: config.secure,
+      ignoreTLS: !config.secure,
+      port: config.port,
+      pool: config.pool,
+      maxConnections: config.maxConnections,
+      maxMessages: config.maxMessages,
+      connectionTimeout: config.connectionTimeout,
+      greetingTimeout: config.greetingTimeout,
+      socketTimeout: config.socketTimeout,
+      dnsTimeout: config.dnsTimeout,
+      sendingRate: this.config.sendingRate,
+      ...auth,
+    };
+    this.emailClient = nodemailer.createTransport(options);
+  }
+
+  async buildHeaders({
+    template,
+    context,
+    headers,
+  }: {
+    template: {
+      name: string;
+      version: number;
+      metricsName?: string;
+    };
+    context: {
+      serverName: string;
+      language: string;
+      deviceId?: string;
+      flowId?: string;
+      flowBeginTime?: number;
+      service?: string;
+      uid?: string;
+      entrypoint?: string;
+      cmsRpClientId?: string;
+    };
+    headers: Record<string, string>;
+  }) {
+    const optionalHeader = (key: string, value?: string | number) => {
+      if (value) {
+        return { [key]: value.toString() };
+      }
+      return {};
+    };
+
+    headers = {
+      'Content-Language': context.language,
+      'X-Template-Name': template.name,
+      'X-Template-Version': template.version.toString(),
+      ...headers,
+      ...optionalHeader('X-Device-Id', context.deviceId),
+      ...optionalHeader('X-Flow-Id', context.flowId),
+      ...optionalHeader('X-Flow-Begin-Time', context.flowBeginTime),
+      ...optionalHeader('X-Service-Id', context.service),
+      ...optionalHeader('X-Uid', context.uid),
+    };
+
+    if (this.config.sesConfigurationSet) {
+      const sesHeaders = [
+        `messageType=fxa-${template.metricsName || template.name}`,
+        'app=fxa',
+        `service=${context.serverName}`,
+        `ses:feedback-id-a=fxa-${template.name}`,
+      ];
+      if (context.cmsRpClientId && context.entrypoint) {
+        sesHeaders.push(`cmsRp=${context.cmsRpClientId}-${context.entrypoint}`);
+      }
+
+      // Note on SES Event Publishing: The X-SES-CONFIGURATION-SET and
+      // X-SES-MESSAGE-TAGS email headers will be stripped by SES from the
+      // actual outgoing email messages.
+      headers['X-SES-CONFIGURATION-SET'] = this.config.sesConfigurationSet;
+      headers['X-SES-MESSAGE-TAGS'] = sesHeaders.join(', ');
+    }
+
+    return headers;
+  }
+
+  async send({
+    to,
+    template,
+    headers,
+    from,
+    cc,
+    subject,
+    preview,
+    text,
+    html,
+  }: Email) {
+    try {
+      await this.bounces.check(to, template);
+    } catch (err) {
+      const tags = { template: template, error: err.errno };
+      this.statsd.increment('email.bounce.limit', tags);
+
+      this.log.error('email.bounce.limit', {
+        err: err.message,
+        errno: err.errno,
+        to,
+        template,
+      });
+      return;
+    }
+
+    this.log.debug('mailer.send', {
```

Comment:
Existing code has `mailer.send.1`, not sure if that's significant

Reply:
There's a follow up about adding retry logic. I wonder if one point there was retry logic? /notsure